python/ray/serve/tests/test_grpc.py

```diff
@@ -4,22 +4,118 @@
 import os
 from ray.serve.drivers import DefaultgRPCDriver, gRPCIngress
 import ray
+import time
 from ray import serve
 from ray.cluster_utils import Cluster
 from ray.serve._private.constants import SERVE_NAMESPACE
+from ray.serve._private.http_proxy import DRAINED_MESSAGE
 from ray._private.test_utils import wait_for_condition, run_string_as_driver
 from ray.serve.exceptions import RayServeException
 
 from ray.serve._private.constants import (
-    SERVE_DEFAULT_APP_NAME,
     DEPLOYMENT_NAME_PREFIX_SEPARATOR,
+    RAY_SERVE_ENABLE_EXPERIMENTAL_STREAMING,
+    SERVE_DEFAULT_APP_NAME,
 )
 
 from unittest.mock import patch
 from ray._private.test_utils import (
     setup_tls,
     teardown_tls,
 )
+from ray.serve.generated import serve_pb2, serve_pb2_grpc
+from ray.serve.tests.test_config_files.grpc_deployment import g, g2
+import grpc
+
+
+def ping_grpc_list_applications(channel, app_names, test_draining=False):
+    stub = serve_pb2_grpc.RayServeAPIServiceStub(channel)
+    request = serve_pb2.ListApplicationsRequest()
+    if test_draining:
+        with pytest.raises(grpc.RpcError) as exception_info:
+            _, _ = stub.ListApplications.with_call(request=request)
+        rpc_error = exception_info.value
+        assert rpc_error.code() == grpc.StatusCode.UNAVAILABLE
+        assert rpc_error.details() == DRAINED_MESSAGE
+    else:
+        response, call = stub.ListApplications.with_call(request=request)
+        assert call.code() == grpc.StatusCode.OK
+        assert response.application_names == app_names
+    return True
```

Comment:
nit: return seems redundant

Reply:
This is one is used in `wait_for_condition` in some cases to ensure the service is up. True will only be returned when all the assertion are passed and signaling condition met. Also see: https://anyscaleteam.slack.com/archives/C043A9QJLNR/p1691772912317919